### PR TITLE
Remove a merge fixme about GetNewOidForAuthId()

### PR DESCRIPTION
**Pull request by adam8157**
Greenplum requires a pre-assigned auth id to keep QD and QEs synchronized in
binary upgrade mode, error out if there is no pre-assigned id as the merge
fixme says.

### CODE REVIEW

**Comment by asimrp** [comment]
PR pipeline is not happy, `pg_upgrade` has failed.  Please take a look, @adam8157

**Comment by adam8157** [comment]
> PR pipeline is not happy, `pg_upgrade` has failed. Please take a look, @adam8157

Thanks for reviewing, now it passed.

File: src/backend/catalog/oid_dispatch.c [hunk 1]
```diff
@@ -350,9 +350,13 @@ GetNewOrPreassignedOid(Relation relation, Oid indexId, AttrNumber oidcolumn,
 		 */
 		if (oid == InvalidOid)
 		{
-			if (IsBinaryUpgrade)
+			if (IS_QUERY_DISPATCHER())
```
Comment (asimrp) [inline src/backend/catalog/oid_dispatch.c:353]: That is confusing.  We are already inside `if (Gp_role == GP_ROLE_EXECUTE || IsBinaryUpgrade)` block.  And the corresponding else block is dedicated to the case `(Gp_role == GP_ROLE_DISPATCH)`.   So something is not right with the proposed code change.

**Review by asimrp — CHANGES_REQUESTED** [review]

File: src/backend/catalog/oid_dispatch.c (hunk 1 above)
Comment (adam8157) [reply src/backend/catalog/oid_dispatch.c:353]: It's `GP_ROLE_UTILITY` and `IsBinaryUpgrade` on the coordinator node here.

`pg_upgrade` establishes the connection as `GP_ROLE_UTILITY`, first on the coordinator node creates the schema, then on the segments retores with the pre-assigned OIDs.

Comment (adam8157) [reply src/backend/catalog/oid_dispatch.c]: I added redundant checks here, `(IS_QUERY_DISPATCHER() && IsBinaryUpgrade && Gp_role == GP_ROLE_UTILITY)`

**Comment by adam8157** [comment]
Have fixed the concern, will merge this the next day if no objection, thanks for the review.

**Pull request merged**